### PR TITLE
Add abifilters to our gradle templates

### LINKED ISSUE #135173: Add `abiFilters` to android gradle templates to prevent serving on x86 devices
Original comment by @chinmaygarde 

>There are no x86 AOT builds supported by Flutter. To disable the application being served on devices those device, the abiFilters directive must be used to only specify the variants Flutter supports. 

We should look in to adding such a validation check ~to the Flutter tools.~

Context: https://github.com/flutter/flutter/issues/83596#issuecomment-852526806

Edited based on comment from @christopherfujino, this should actually be done by adding `abiFilters` to our gradle templates: https://developer.android.com/ndk/guides/abis#gc

### PR DESCRIPTION
**Pull request by gmackall**
Fixes https://github.com/flutter/flutter/issues/135173, and could also be interpreted as fixing https://github.com/flutter/flutter/issues/83596 based on @chinmaygarde 's comment.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by gmackall** [comment]
cc @chinmaygarde , as you made the original comment - does this look like the correct list to you, based on the table at https://developer.android.com/ndk/guides/abis#gradle?

File: packages/flutter_tools/templates/app_shared/android-java.tmpl/app/build.gradle.tmpl
Comment (gmackall) [inline packages/flutter_tools/templates/app_shared/android-java.tmpl/app/build.gradle.tmpl:45]: This is the list the above comment was referring to

**Review by johnmccutchan — APPROVED** [review]

**Comment by christopherfujino** [comment]
@gmackall is this pending Chinmay's review, or is it good to go once presubmits go green?

**Comment by gmackall** [comment]
> @gmackall is this pending Chinmay's review, or is it good to go once presubmits go green?

I would want confirmation from Chinmay here before landing

**Review by chinmaygarde — APPROVED** [review]
Oh, I didn't know this was waiting on my review. This list LGTM. I think the x86 variant has to be created separately is needed and can't do AOT either.

**Comment by andrewkolos** [comment]
@gmackall Is this ready to be merged?

**Pull request merged**
